config-service-impl/src/main/java/org/hypertrace/config/service/store/ConstantExpressionConverter.java

```diff
@@ -0,0 +1,48 @@
+package org.hypertrace.config.service.store;
+
+import com.google.protobuf.Value;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
+
+public class ConstantExpressionConverter {
+
+  public static ConstantExpression fromProtoValue(Value value) {
+    switch (value.getKindCase()) {
+      case STRING_VALUE:
+        return ConstantExpression.of(value.getStringValue());
+      case NUMBER_VALUE:
+        return ConstantExpression.of(value.getNumberValue());
+      case BOOL_VALUE:
+        return ConstantExpression.of(value.getBoolValue());
+      case LIST_VALUE:
+        List<Value> values = value.getListValue().getValuesList();
+        // Infer and cast list element types (assumes homogeneous list)
```

Comment:
Rather than assume it and leave a comment we can easily validate this. e.g. collect kinds to set and ensure `== 1`. Or really `<= 1`,  why not support empty list? Seems trivial to do and there are reasonable use cases (e.g. `my_array_field EQ []`)

Reply:
Fixed it